src/internal/backoff/exponential.go

```diff
@@ -0,0 +1,51 @@
+package backoff
+
+import (
+	"sync/atomic"
+	"time"
+)
+
+// ExponentialBackoff is a struct that handles backoff duration.
+type ExponentialBackoff struct {
```

Comment:
You are repeating the name of the package here. When used it will sound `backoff.ExponentialBackoff` and `backoff.Exponential` would be enough.